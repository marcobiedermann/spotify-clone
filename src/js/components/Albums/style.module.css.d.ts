declare const styles: {
  readonly albums: string;
};
export = styles;
