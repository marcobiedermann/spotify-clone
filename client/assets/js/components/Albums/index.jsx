import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import Album from '../Album';
import styles from './style.css';

class Albums extends PureComponent {
  static propTypes = {
    albums: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
      }),
    ),
  };

  static defaultProps = {
    albums: [],
  };

  render() {
    const { albums } = this.props;

    return (
      <ul className={styles.albums}>
        {albums.map(album => (
          <li key={album.id}>
            <Album {...album} />
          </li>
        ))}
      </ul>
    );
  }
}

export default Albums;
