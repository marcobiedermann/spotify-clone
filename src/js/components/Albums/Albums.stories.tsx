import type { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Albums from '.';
import albumFixtures from '../Album/__fixtures__';

const meta: Meta<typeof Albums> = {
  component: Albums,
  args: {
    items: [albumFixtures],
  },
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
};

type Story = StoryObj<typeof Albums>;

const Default: Story = {};

export { Default };
export default meta;
