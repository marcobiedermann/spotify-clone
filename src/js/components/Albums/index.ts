import Albums from './Albums';

export default Albums;
export * from './Albums';
