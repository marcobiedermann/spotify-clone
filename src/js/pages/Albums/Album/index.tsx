import React from 'react';
import { Helmet } from 'react-helmet';
import { useParams } from 'react-router-dom';
import { Album, Error, Loader } from '../../../components';
import { useAlbum } from '../../../hooks/albums';

function AlbumPage(): JSX.Element {
  const { albumId } = useParams();
  const { data, error, isError, isLoading } = useAlbum(albumId!);

  if (isError) {
    return <Error>{error.message}</Error>;
  }

  if (isLoading) {
    return <Loader />;
  }

  const { name } = data;

  return (
    <>
      <Helmet>
        <title>{name}</title>
      </Helmet>
      <Album {...data} />
    </>
  );
}

export default AlbumPage;
