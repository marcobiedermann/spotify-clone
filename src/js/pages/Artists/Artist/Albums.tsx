/* eslint-disable jsx-a11y/label-has-associated-control */

import React from 'react';
import { useForm } from 'react-hook-form';
import { Link, useLocation, useParams } from 'react-router-dom';
import { useArtistAlbums } from '../../../hooks/artists';
import { Albums, Error, Loader, Section } from '../../../components';

interface FormData {
  include_group: 'album' | 'single';
}

function Discography() {
  const { register, watch } = useForm<FormData>({
    defaultValues: {
      include_group: 'album',
    },
  });
  const include_group = watch('include_group');
  const { pathname } = useLocation();
  const { artistId } = useParams();
  const { data, error, isError, isPending } = useArtistAlbums(artistId!, {
    include_groups: [include_group],
    limit: 9,
  });

  if (isError) {
    return <Error>{error.message}</Error>;
  }

  if (isPending) {
    return <Loader />;
  }

  const { items } = data;

  return (
    <Section>
      <div
        style={{
          alignItems: 'center',
          display: 'flex',
          gap: '1em',
          justifyContent: 'space-between',
        }}
      >
        <h2>Discography</h2>
        <p>
          <Link to={`${pathname}/albums`}>Show all</Link>
        </p>
      </div>
      <form>
        <div>
          <label htmlFor="album">Album</label>
          <input id="album" type="radio" value="album" {...register('include_group')} />
        </div>
        <div>
          <label htmlFor="single">Single</label>
          <input id="single" type="radio" value="single" {...register('include_group')} />
        </div>
      </form>
      <Albums items={items} />
    </Section>
  );
}

export default Discography;
