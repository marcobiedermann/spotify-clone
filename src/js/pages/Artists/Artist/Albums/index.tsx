import React from 'react';
import { Helmet } from 'react-helmet';
import { useParams } from 'react-router-dom';
import { Albums, Error, Loader } from '../../../../components';
import { useArtistAlbums } from '../../../../hooks/artists';

function AlbumsPage(): JSX.Element {
  const { artistId } = useParams();
  const { data, error, isError, isPending } = useArtistAlbums(artistId!);

  if (isError) {
    return <Error>{error.message}</Error>;
  }

  if (isPending) {
    return <Loader />;
  }

  const { items } = data;

  return (
    <>
      <Helmet>
        <title>Albums</title>
      </Helmet>
      <Albums items={items} />
    </>
  );
}

export default AlbumsPage;
