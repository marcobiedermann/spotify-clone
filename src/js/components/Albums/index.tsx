import classNames from 'classnames';
import React from 'react';
import Album, { AlbumProps } from '../Album';
import styles from './style.module.css';

export interface AlbumsProps {
  className?: string;
  items?: AlbumProps[];
}

function Albums(props: AlbumsProps): JSX.Element {
  const { className, items = [] } = props;

  return (
    <ul className={classNames(className, styles.albums)}>
      {items.map((album) => (
        <li key={album.id}>
          <Album {...album} />
        </li>
      ))}
    </ul>
  );
}

export default Albums;
