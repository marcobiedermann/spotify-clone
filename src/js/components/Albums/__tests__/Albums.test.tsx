import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';
import Albums from '..';

describe('Albums component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Albums items={[]} />
      </MemoryRouter>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
