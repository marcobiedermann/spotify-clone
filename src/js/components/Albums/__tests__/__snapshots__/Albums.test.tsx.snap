// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Albums component > renders correctly 1`] = `
<ul
  class="albums"
>
  <li>
    <div>
      <figure>
        <a
          href="/albums/2up3OPMp9Tb4dAKM2erWXQ"
        >
          <img
            alt="string"
            height="300"
            src="https://i.scdn.co/image/ab67616d00001e02ff9ca10b55ce82ae553c8228"
            width="300"
          />
        </a>
        <figcaption>
          <h3>
            <a
              href="/albums/2up3OPMp9Tb4dAKM2erWXQ"
            >
              string
            </a>
          </h3>
          <h4>
            <a
              href="/artists/string"
            >
              string
            </a>
          </h4>
        </figcaption>
      </figure>
      <table>
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Title
            </th>
            <th
              align="right"
            >
              Duration
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              0
            </td>
            <td>
              <div
                class="track__name"
              >
                string
              </div>
              <div>
                <a
                  href="/artists/string"
                >
                  string
                </a>
              </div>
            </td>
            <td
              align="right"
            >
              0:00
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </li>
</ul>
`;
